### PR TITLE
[ADD][11.0] Adding module web_widget_image_url

### PR DESCRIPTION
**Pull request by pnajman-modoolar**
Hello,

Module ``web_widget_image_url`` provides web widget for displaying image from URL on form and tree views.

Kind Regards,
Petar

### CODE REVIEW

File: web_widget_image_url/README.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
Comment (elicoidal) [inline web_widget_image_url/README.rst]: https

File: web_widget_image_url/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Copyright 2017 - 2018 Modoolar <info@modoolar.com>
```
Comment (elicoidal) [inline web_widget_image_url/__init__.py]: usually copyright and licenses not needed in `__init__.py`

File: web_widget_image_url/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2017 - 2018 Modoolar <info@modoolar.com>
+# License LGPLv3.0 or later (https://www.gnu.org/licenses/lgpl-3.0.en.html).
+{
+    "name": "Web Image URL",
+    "summary": "This module provides web widget for displaying image from URL",
+    "category": "Web",
+    "version": "11.0.1.0.0",
+    "license": "LGPL-3",
+    "author": "Odoo Community Association (OCA), Modoolar",
+    "website": "https://www.modoolar.com/",
```
Comment (elicoidal) [inline web_widget_image_url/__manifest__.py]: should be the URL of the github repo: https://github.com/OCA/web/

File: web_widget_image_url/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    "depends": ["web"],
+    "data": [
+        "views/web_widget_image_url.xml",
+    ],
+    "qweb": ["static/src/xml/*.xml"],
+    "images": [],
```
Comment (elicoidal) [inline web_widget_image_url/__manifest__.py]: remove empty key or better add a cover page

**Review by elicoidal — CHANGES_REQUESTED** [review]
Almost perfect. Thanks for the contribution.
No technical review.
I would suggest to add some demo data (but non blocking on my side

File: web_widget_image_url/README.rst [hunk 5]
```diff
@@ -0,0 +1,44 @@
+.. image:: https://www.gnu.org/graphics/lgplv3-147x51.png
+   :target: https://www.gnu.org/licenses/lgpl-3.0.en.html
+   :alt: License: LGPL-v3
+
+==============
```
Comment (simahawk) [inline web_widget_image_url/README.rst]: RST validators will complain: `=` length  should be equal to wrapped text length

File: web_widget_image_url/README.rst [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+Usage
+=====
+
+To use this module, you need to decorate xml field with widget ``image_url``.
+
+
+.. code-block:: xml
+
+    <field name="avatar" widget="image_url" class="oe_image_small" width="100px" height="100px"/>
```
Comment (simahawk) [inline web_widget_image_url/README.rst:19]: would be nice to see a screenshot here.

File: web_widget_image_url/README.rst [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
```
Comment (simahawk) [inline web_widget_image_url/README.rst:20]: you miss the issue section. See default template https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst#known-issues--roadmap

File: web_widget_image_url/__manifest__.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2017 - 2018 Modoolar <info@modoolar.com>
+# License LGPLv3.0 or later (https://www.gnu.org/licenses/lgpl-3.0.en.html).
+{
+    "name": "Web Image URL",
+    "summary": "This module provides web widget for displaying image from URL",
+    "category": "Web",
+    "version": "11.0.1.0.0",
+    "license": "LGPL-3",
+    "author": "Odoo Community Association (OCA), Modoolar",
```
Comment (simahawk) [inline web_widget_image_url/__manifest__.py]: Author first, OCA should be the last one.

File: web_widget_image_url/static/src/xml/web_widget_image_url.xml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+-->
+<templates xml:space="preserve">
+<t t-name="FieldImageURL">
+    <span class="oe_form_field oe_form_field_image" t-att-style="widget.attrs.style">
+        <img/>
+    </span>
+</t>
+
+<t t-name="FieldImageURL-img">
+    <img t-att-src='url'
```
Comment (simahawk) [inline web_widget_image_url/static/src/xml/web_widget_image_url.xml]: do not mix up single/double quotes

File: web_widget_image_url/static/src/xml/web_widget_image_url.xml [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+        t-att-height="widget.attrs.img_height || widget.attrs.height"
+    />
+</t>
+
+<img t-name="ListView.row.img.url" t-att-src='url'
+        t-att-border="0"
+        t-att-name="widget.name"
+        t-att-width="width"
+        t-att-height="height">
+</img>
```
Comment (simahawk) [inline web_widget_image_url/static/src/xml/web_widget_image_url.xml]: `img` should be self-closed

File: web_widget_image_url/static/src/js/web_widget_image_url.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+        className: 'o_attachment_image',
+        template: 'FieldImageURL',
+        placeholder: "/web/static/src/img/placeholder.png",
+        supportedFieldTypes: ['char'],
+
+        _render: function () {
+            var self = this;
+            var url = this.value ? this.value : this.placeholder;
+
+            var $img = $(QWeb.render("FieldImageURL-img", {widget: this, url: url}));
```
Comment (simahawk) [inline web_widget_image_url/static/src/js/web_widget_image_url.js]: I don't think you need all this code :thinking: 
Why don't you rely on standard rendering? `FieldImageURL-img` content should stay inside `FieldImageURL` template and you could override `_render` to call `_super` and then just hook the error event.
Am I missing anything?

File: web_widget_image_url/static/src/xml/web_widget_image_url.xml [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+<t t-name="FieldImageURL-img">
+    <img t-att-src='url'
+        t-att-border="widget.readonly ? 0 : 1"
+        t-att-name="widget.name"
+        t-att-width="widget.attrs.img_width || widget.attrs.width"
+        t-att-height="widget.attrs.img_height || widget.attrs.height"
+    />
+</t>
+
+<img t-name="ListView.row.img.url" t-att-src='url'
```
Comment (simahawk) [inline web_widget_image_url/static/src/xml/web_widget_image_url.xml]: Where is this used?

**Review by simahawk — CHANGES_REQUESTED** [review]

File: web_widget_image_url/README.rst (hunk 1 above)
Comment (pnajman-modoolar) [reply web_widget_image_url/README.rst]: Fixed

File: web_widget_image_url/__init__.py (hunk 2 above)
Comment (pnajman-modoolar) [reply web_widget_image_url/__init__.py]: Done

File: web_widget_image_url/__manifest__.py (hunk 3 above)
Comment (pnajman-modoolar) [reply web_widget_image_url/__manifest__.py]: Done

File: web_widget_image_url/README.rst (hunk 5 above)
Comment (pnajman-modoolar) [reply web_widget_image_url/README.rst]: Fixed

File: web_widget_image_url/README.rst (hunk 7 above)
Comment (pnajman-modoolar) [reply web_widget_image_url/README.rst:20]: Fixed

File: web_widget_image_url/__manifest__.py (hunk 8 above)
Comment (pnajman-modoolar) [reply web_widget_image_url/__manifest__.py]: Done

File: web_widget_image_url/static/src/xml/web_widget_image_url.xml (hunk 9 above)
Comment (pnajman-modoolar) [reply web_widget_image_url/static/src/xml/web_widget_image_url.xml]: Fixed

File: web_widget_image_url/static/src/xml/web_widget_image_url.xml (hunk 10 above)
Comment (pnajman-modoolar) [reply web_widget_image_url/static/src/xml/web_widget_image_url.xml]: Fixed

File: web_widget_image_url/static/src/xml/web_widget_image_url.xml (hunk 12 above)
Comment (pnajman-modoolar) [reply web_widget_image_url/static/src/xml/web_widget_image_url.xml]: Removed

**Comment by pnajman-modoolar** [comment]
If all is good, can we merge this branch with 11.0?

**Comment by pedrobaeza** [comment]
There are still comments unattended and 2 positive reviews are needed.

File: web_widget_image_url/__manifest__.py (hunk 4 above)
Comment (pnajman-modoolar) [reply web_widget_image_url/__manifest__.py]: Removed

File: web_widget_image_url/static/src/js/web_widget_image_url.js (hunk 11 above)
Comment (pnajman-modoolar) [reply web_widget_image_url/static/src/js/web_widget_image_url.js]: You are right. It's fixed now.

File: web_widget_image_url/README.rst (hunk 6 above)
Comment (pnajman-modoolar) [reply web_widget_image_url/README.rst:19]: Screenshot added :)

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM :+1:

**Comment by pnajman-modoolar** [comment]
@simahawk, @elicoidal, we need one more positive review :)

**Review by simahawk — APPROVED** [review]

**Comment by pnajman-modoolar** [comment]
@pedrobaeza, @elicoidal, we have two positive reviews :)

**Pull request merged**